Set  .net compilation output stream to utf8

dream-young-soul opened the pull request:
Set the .net compilation output stream to utf8 to prevent garbled characters in other languages


![20230224145959](https://user-images.githubusercontent.com/17066147/221117819-1ea39c2d-34bf-4422-8eac-c9941e950230.png)

dream-young-soul committed:
.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

Set .net compilation output stream to utf8

Set .net compilation output stream to utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

Set .net compilation output stream to utf8

dream-young-soul committed:
.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

dream-young-soul committed:
.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

Set .net compilation output stream to utf8

Set .net compilation output stream to utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

dream-young-soul committed:
.net build_log utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

dream-young-soul committed:
Set .net compilation output stream to utf8

akien-mga commented on modules/mono/editor/GodotTools/GodotTools/Build/BuildSystem.cs:
```diff
@@ -50,7 +50,7 @@ public static class BuildSystem
             if (stdErrHandler != null)
                 process.ErrorDataReceived += (_, e) => stdErrHandler.Invoke(e.Data);
-
+            process.StartInfo.StandardOutputEncoding = Encoding.UTF8;
```
This would be better defined together with other `startInfo` properties before it's passed to the `Process` constructor:
```
            startInfo.RedirectStandardOutput = true;
            startInfo.RedirectStandardError = true;
            startInfo.UseShellExecute = false;
            startInfo.CreateNoWindow = true;
            startInfo.StandardOutputEncoding = Encoding.UTF8;
            startInfo.EnvironmentVariables["DOTNET_CLI_UI_LANGUAGE"]
                = ((string)editorSettings.GetSetting("interface/editor/editor_language")).Replace('_', '-');
```

Same for the other one below.

akien-mga commented on modules/mono/editor/GodotTools/GodotTools/Build/DotNetFinder.cs:
```diff
@@ -61,7 +62,7 @@ public static string FindDotNetExe()
 
             var lines = new List<string>();
-
+            process.StartInfo.StandardOutputEncoding = Encoding.UTF8;
```
Same as for the other file, define this directly in the `ProcessStartInfo` constructor instead of setting it up after the fact and between the definition and use of `lines`.
```
using Process process = new Process();
            process.StartInfo = new ProcessStartInfo(dotNetExe, "--list-sdks")
            {
                UseShellExecute = false,
                RedirectStandardOutput = true,
                StandardOutputEncoding = Encoding.UTF8
            };
```

akien-mga commented:
Could you squash the commits? And make sure to use the more explicit commit message for the squashed commit (like "Set .NET compilation output stream to UTF-8". See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

dream-young-soul commented:
> 你能压缩提交吗？并确保为压缩的提交使用更明确的提交消息（例如“将 .NET 编译输出流设置为 UTF-8”。有关说明，请参阅[PR 工作流程。](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

OK, I'll give it a try. This is my first PR submission. Many processes do not understand

akien-mga commented:
Let me know if you need help. I can also update the PR myself, but if you're interested it's a good opportunity to learn Godot's PR workflow.

dream-young-soul committed:
Merge branch 'master' of https://github.com/dream-young-soul/godot

dream-young-soul committed:
Set .net compilation output stream to utf8

Set the .net compilation output stream to utf8 to prevent garbled characters in other languages

dream-young-soul commented:
> Let me know if you need help. I can also update the PR myself, but if you're interested it's a good opportunity to learn Godot's PR workflow and some Git tricks.

I changed the code and tried to merge, was it successful?

akien-mga commented:
After squashing the commits, you should actually have force pushed (`git push --force`) instead of merging your remote `master` branch back into the local one. Now there are 5 commits when there should be one.

Here's a branch with the commits squashed and a couple unnecessary formatting changes removed:
https://github.com/akien-mga/godot/tree/dream-young-soul/dotnet-process-startinfo-utf8

dream-young-soul committed:
Merge branch 'master' of https://github.com/dream-young-soul/godot

dream-young-soul committed:
Merge branch 'master' of https://github.com/dream-young-soul/godot

dream-young-soul commented:
> After squashing the commits, you should actually have force pushed (`git push --force`) instead of merging your remote `master` branch back into the local one. Now there are 5 commits when there should be one.
> 
> Here's a branch with the commits squashed and a couple unnecessary formatting changes removed: https://github.com/akien-mga/godot/tree/dream-young-soul/dotnet-process-startinfo-utf8

It's a bit of a crash, and the number of submissions is increasing. Please help me with this PR. I should learn how to use github

akien-mga commented:
I can't update this PR it seems, probably you disabled this option when making the PR:
![image](https://user-images.githubusercontent.com/4701338/221154225-137befab-885c-42a6-8831-898b633c59c9.png)

I'll make a new PR with those changes.

akien-mga commented:
Superseded by #73865.

dream-young-soul commented:
> [由#73865](https://github.com/godotengine/godot/pull/73865)取代。

Thanks
